core-bundle/src/Routing/RouteProvider.php

```diff
@@ -0,0 +1,472 @@
+        }
+
+        if (0 === strncmp($pathInfo, '/', 1)) {
+            $pathInfo = substr($pathInfo, 1);
+        }
+
+        if ($this->configAdapter->get('addLanguageToUrl')) {
+            $matches = [];
+
+            if (preg_match('@^([a-z]{2}(-[A-Z]{2})?)/(.+)$@', $pathInfo, $matches)) {
+                $pathInfo = $matches[3];
+            } else {
+                return null;
+            }
+        }
+
+        return $pathInfo;
+    }
+
+    /**
+     * Compile all possible aliases by applying dirname() to the request (e.g. news/archive/item, news/archive, news).
+     *
+     *
+     * @return array
+     */
+    private function getAliasCandidates(string $pathInfo)
+    {
+        $pos = strpos($pathInfo, '/');
+
+        if (false === $pos) {
+            return [$pathInfo];
+        }
+
+        if (!$this->configAdapter->get('folderUrl')) {
+            return [substr($pathInfo, 0, $pos)];
+        }
+
+        $candidates = [$pathInfo];
+
+        while ('/' !== $pathInfo && false !== strpos($pathInfo, '/')) {
+            $pathInfo = \dirname($pathInfo);
+            $candidates[] = $pathInfo;
+        }
+
+        return $candidates;
+    }
+
+    private function addRoutesForPages(iterable $pages, array &$routes): void
+    {
+        /** @var PageModel $page */
+        foreach ($pages as $page) {
+            $this->addRoutesForPage($page, $routes);
+        }
+    }
+
+    private function addRoutesForRootPages(array $pages, array &$routes): void
+    {
+        if (null === $pages) {
+            return;
+        }
+
+        /** @var PageModel $page */
+        foreach ($pages as $page) {
+            $this->addRoutesForRootPage($page, $routes);
+        }
+    }
+
+    private function createCollectionForRoutes(array $routes, array $languages): RouteCollection
+    {
+        $this->sortRoutes($routes, $languages);
+
+        $collection = new RouteCollection();
+
+        foreach ($routes as $name => $route) {
+            $collection->add($name, $route);
+        }
+
+        return $collection;
+    }
+
+    private function addRoutesForPage(PageModel $page, array &$routes): void
+    {
+        $page->loadDetails();
+
+        $defaults = $this->getRouteDefaults($page);
+        $defaults['parameters'] = '';
+        $requirements = ['parameters' => '(/.+)?'];
+
+        $path = sprintf('/%s{parameters}%s', $page->alias ?: $page->id, $this->configAdapter->get('urlSuffix'));
+        $table = $this->pageAdapter->getTable();
+
+        if ($this->configAdapter->get('addLanguageToUrl')) {
+            $path = '/{_locale}'.$path;
+            $requirements['_locale'] = $page->rootLanguage;
+        }
+
+        $routes[$table.'.'.$page->id] = new Route(
+            $path,
+            $defaults,
+            $requirements,
+            ['utf8' => true],
+            $page->domain,
+            null // TODO should we match SSL only if enabled in root? => $page->rootUseSSL ? 'https' : null
```

Comment:
I would prefer to automatically redirect HTTP to HTTPS.

Reply:
see #262